data_steward/cdr_cleaner/cleaning_rules/drop_multiple_measurements.py

```diff
@@ -10,89 +10,136 @@
 
-INTERMEDIARY_TABLE = 'DC617_dropped_mult_measurements'
+ISSUE_NUMBERS = ['DC847']
+MEASUREMENT = 'measurement'
 
-INVALID_MULT_MEASUREMENTS = """
+SANDBOX_INVALID_MULT_MEASUREMENTS = JINJA_ENV.from_string("""
 CREATE OR REPLACE TABLE
-  `{project}.{sandbox_dataset}.{intermediary_table}` AS (
+  `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` AS 
 SELECT
   *
 FROM
 (SELECT *, ROW_NUMBER() OVER(PARTITION BY person_id, measurement_source_concept_id ORDER BY
       measurement_datetime DESC) AS row_num
- FROM `{project}.{dataset}.measurement`
+ FROM `{{project}}.{{dataset}}.measurement`
  WHERE measurement_source_concept_id IN (903131,903119,903107,903124,903115,903126,903136,903118,903135,903132,
                                          903110,903112,903117,903109,903127,1586218,903133,903111,903120,903113,
                                          903129,903105,903125,903114,903134,903116,903106,903108,903123,903130,
                                          903128,903122,903121)
  ORDER BY person_id, measurement_source_concept_id, row_num)
---selection measurements to delete where row_num !=1
 WHERE row_num != 1
-)
-"""
+""")
 
-VALID_MEASUREMENTS = """
+REMOVE_INVALID_MULT_MEASUREMENTS = JINJA_ENV.from_string("""
 DELETE FROM
-  `{project}.{dataset}.measurement`
+  `{{project}}.{{dataset}}.measurement`
 WHERE
 (person_id, measurement_concept_id, measurement_id, measurement_date, measurement_datetime)
 IN( SELECT
     (person_id, measurement_concept_id, measurement_id, measurement_date, measurement_datetime)
-    FROM `{project}.{sandbox_dataset}.{intermediary_table}` )
-"""
+    FROM `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` )
+""")
 
 
-def get_drop_multiple_measurement_queries(project_id, dataset_id,
-                                          sandbox_dataset_id):
+class DropMultipleMeasurements(BaseCleaningRule):
     """
-    runs the query which removes all multiple me
-
-    :param project_id: Name of the project
-    :param dataset_id: Name of the dataset where the queries should be run
-    :return:
+    Removes all but the most recent of each Physical Measurement for all participants.
     """
-    queries_list = []
-
-    invalid_measurements_query = dict()
-    invalid_measurements_query[
-        cdr_consts.QUERY] = INVALID_MULT_MEASUREMENTS.format(
-            dataset=dataset_id,
-            project=project_id,
-            sandbox_dataset=sandbox_dataset_id,
-            intermediary_table=INTERMEDIARY_TABLE)
-    queries_list.append(invalid_measurements_query)
-
-    valid_measurements_query = dict()
-    valid_measurements_query[cdr_consts.QUERY] = VALID_MEASUREMENTS.format(
-        dataset=dataset_id,
-        project=project_id,
-        sandbox_dataset=sandbox_dataset_id,
-        intermediary_table=INTERMEDIARY_TABLE)
-    queries_list.append(valid_measurements_query)
 
-    return queries_list
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
```

Comment:
One more update: I think `table_namer=None` should be a parameter passed to super.

Reply:
Updated!

Reply:
Sorry, I should've clarified: This parameter should be on line 57 in the initialization of the class as well. This will prevent the warning in clean_cdr_engine: line 220 from firing.

Reply:
Got it - thanks, Nishanth!